multicast/src/main/kotlin/com/dropbox/flow/multicast/Multicaster.kt

```diff
@@ -72,21 +75,42 @@ class Multicaster<T>(
         )
     }
 
+    /**
+     * The shared downstream flow. Collectors of this flow will share values dispatched by the
+     * [source] Flow.
+     */
     val flow = flow<T> {
         val channel = Channel<ChannelManager.Message.Dispatch.Value<T>>(Channel.UNLIMITED)
         val subFlow = channel.consumeAsFlow()
-                .onStart {
+            .onStart {
+                try {
```

Comment:
Does this look cleaner? I think `ClosedSendChannelException` is the only possible exception that can be thrown here as the `actor` in `ChannelManager` always closes without a `cause`.
```suggestion
                runCatching {
                    channelManager.addDownstream(channel)
                }.onFailure {
                    // before we could start, channel manager was closed.
                    // close our downstream manually as it won't be closed by the ChannelManager
                    channel.close()
                }
```